biorxiv/article_distances/biorxiv_article_distances.py

```diff
@@ -0,0 +1,499 @@
+    )
+)
+articles_distance_df.head()
+
+
+# ## biorxiv -> random paper same journal
+
+# In[13]:
+
+
+PMC_off_published = (
+    pmc_journal_mapped_df
+    .drop("doi", axis=1)
+    .query(f"document not in {biorxiv_published.reset_index().pmcid.tolist()}")
+    .query(f"journal in {articles_distance_df.journal.unique().tolist()}")
+    .groupby("journal", group_keys=False)
+    .apply(lambda x: x.sample(1, random_state=100))
+    .set_index("document")
+)
+PMC_off_published.head()
+
+
+# In[14]:
+
+
+journal_mapper = {
+    journal:col
+    for col, journal in enumerate(PMC_off_published.journal.tolist())
+}
+list(journal_mapper.items())[0:10]
+
+
+# In[17]:
+
+
+off_article_dist = cdist(
+    biorxiv_published
+    .loc[PMC_pubished.index.tolist()]
+    .drop(["document", "published_doi", "pmcoa"], axis=1), 
+    PMC_off_published.drop("journal", axis=1),
+    'euclidean'
+)
+off_article_dist.shape
+
+
+# In[18]:
+
+
+data = []
+for idx, row in tqdm.tqdm(articles_distance_df.iterrows()):
+    if row['journal'] in journal_mapper:
+        data.append(
+            {
+                "document": row['document'],
+                "pmcid":  (
+                    PMC_off_published
+                    .query(f"journal=='{row['journal']}'")
+                    .reset_index()
+                    .document
+                    .values[0]
+                ),
+                "journal": row['journal'],
+                "distance": off_article_dist[idx, journal_mapper[row['journal']]]
+            }
+        )
+
+
+# In[19]:
+
+
+final_df = (
+    articles_distance_df
+    .assign(label="pre_vs_published")
+    .append(
+        pd.DataFrame
+        .from_records(data)
+        .assign(label="pre_vs_random")
+    )
+)
+final_df.head()
+
+
+# In[20]:
+
+
+final_df = (
+    biorxiv_journal_df[["document", "doi"]]
+    .merge(final_df)
+)
+final_df.to_csv("output/article_distances.tsv", sep="\t", index=False)
+final_df.head()
+
+
+# # Distribution plot
+
+# In[21]:
+
+
+g = (
+    p9.ggplot(final_df)
+    + p9.aes(x="label",y="distance")
+    + p9.geom_boxplot()
+    + p9.coord_flip()
+    + p9.theme_seaborn()
+)
+g.save("output/biorxiv_article_distance.svg", dpi=500)
+g.save("output/biorxiv_article_distance.png", dpi=500)
+print(g)
+
+
+# # Find bioRxiv unpublished ->  published PMC articles
+
+# In[31]:
+
+
+biorxiv_unpublished = (
+    biorxiv_journal_mapped_df
+    .query("published_doi.isnull()")
+    .drop(["published_doi", "pmcid", "pmcoa"], axis=1)
+)
+print(biorxiv_unpublished.shape)
+biorxiv_unpublished.head()
+
+
+# In[23]:
+
+
+PMC_unlinked = (
+    pmc_journal_mapped_df
+    .query(
+        f"""
+        document not in {
+            biorxiv_published
+            .reset_index()
+            .pmcid
+            .tolist()
+        }
+        """
+    )
+)
+print(PMC_unlinked.shape)
+PMC_unlinked.head()
+
+
+# In[24]:
+
+
+cutoff_score = (
+    final_df
+    .query("label=='pre_vs_random'")
+    .distance
+    .min()
+)
```

Comment:
If you use regularized logistic regression on the differences between the articles you may be able to get more separation between `pre_vs_random` and `pre_vs_published`. Right now the model is operating under the assumption that all embedded dimensions are equally important for determining whether two articles are the same, which may not be true.

Reply:
Along similar lines, I was going to ask if you tried with fewer dimensions since 300 might be hitting the curse of dimensionality where distances are equal between points.